src/test/java/persistence/sql/dml/SelectQueryBuilderTest.java

```diff
@@ -38,6 +39,19 @@ void findByStringIdQuery() {
         SelectQueryBuilder selectQueryBuilder = new SelectQueryBuilder();
         String findByIdQuery = selectQueryBuilder.findById(metadata.getEntityTable(), metadata.getEntityColumns(), "yang");
 
-        assertEquals(findByIdQuery, "select id, nick_name, old, email FROM users where id = 'yang'");
+        assertEquals(findByIdQuery, "select users.id, users.nick_name, users.old, users.email FROM users where id = 'yang'");
+    }
+
+    @Test
+    @DisplayName("Person 객체로 Select(findById) Query 만들기")
+    void findAllWithJoinQuery() {
+        Metadata mainMetaData = new Metadata(Order.class);
+        Metadata joinMetaData = new Metadata(OrderItem.class);
+
+        SelectQueryBuilder selectQueryBuilder = new SelectQueryBuilder();
+        String findByIdQuery = selectQueryBuilder.findAllWithJoin(mainMetaData.getEntityTable(), mainMetaData.getEntityColumns(),
+                joinMetaData.getEntityTable(), joinMetaData.getEntityColumns());
+
+        assertEquals(findByIdQuery, "SELECT orders.id, orders.orderNumber, orders.orderItems, order_items.id, order_items.product, order_items.quantity FROM orders LEFT JOIN order_items ON orders.order_id = order_items.id");
```

Comment:
Left join 구문을 잘 만들어 주셨네요. 가독성을 위해서 Java15 에 추가된 [Text block](https://www.baeldung.com/java-text-blocks) 을 이용해봐도 좋을 것 같아요 :)

```suggestion
        SelectQueryBuilder selectQueryBuilder = new SelectQueryBuilder();
        String findByIdQuery = selectQueryBuilder.findAllWithJoin(mainMetaData.getEntityTable(), mainMetaData.getEntityColumns(),
                joinMetaData.getEntityTable(), joinMetaData.getEntityColumns());

        assertEquals(findByIdQuery, """
                SELECT orders.id, orders.orderNumber, orders.orderItems, order_items.id, order_items.product, order_items.quantity \
                FROM orders \
                LEFT JOIN order_items ON orders.order_id = order_items.id\
                """
        );
    }
```

Reply:
확실히 텍스트블록을 사용하니 쿼리가 한눈에 들어와서 가독성이 좋네요 👍